src/log-entry-dashboard/log-entry-form/log-entry-form.component.tsx

```diff
@@ -20,124 +20,305 @@ import {
+    this.setState({ categoryId: e });
+  };
+
+  handleRaidoChange = (e: RadioChangeEvent) => {
+    if (e.target.value === 'completed') {
+      this.setState({ pointsCompeleted: 'completed' });
+    } else if (e.target.value === 'notCompleted') {
+      this.setState({ pointsCompeleted: 'notCompleted' });
+    }
+  };
+
+  hasErrors = (fieldsError: any) => {
+    return Object.keys(fieldsError).some(field => fieldsError[field]);
+  };
+
   handleOnSubmit = (e: React.SyntheticEvent) => {
     e.preventDefault();
     const {
       authContext: { currentUser },
       form,
     } = this.props;
+
     form.validateFields((err, values) => {
       if (err) {
         return;
       }
 
       createLogEntry({
-        title: values['description'],
-        arePointsCompleted: values['completedPoints'],
+        title: values['title'],
+        arePointsCompleted: values['completedPoints'] || null,
         notes: values['notes'],
         date: values['logDate'],
         value: values['logValue'],
         categoryId: values['category'], //Categories.BillableTime,
-        subCategoryId: values['subcategory'], //SubCategories.Admin,
+        subCategoryId: values['subcategory'] || null, //SubCategories.Admin,
         updatedBy: currentUser ? currentUser.id : '',
         createdAt: new Date(),
         updatedAt: new Date(),
       });
+
+      form.resetFields();
     });
   };
 
-  render() {
+  handleClearForm = () => {
+    const { form } = this.props;
+    form.resetFields();
+  };
+
+  renderCategoriesSelectOptions = () => {
+    let keys = Object.keys(Categories).filter(
+      (key: any) => !isNaN(Number(Categories[key]))
+    );
+
+    return keys.map((category: any) => {
+      return (
+        <SelectOption key={category} value={Categories[category]}>
+          {category}
+        </SelectOption>
+      );
+    });
+  };
+
+  renderSubCategoriesSelectOptions = () => {
+    let keys = Object.keys(SubCategories).filter(
+      (key: any) => !isNaN(Number(SubCategories[key]))
+    );
+
+    return keys.map((subCategory: any) => {
+      return (
+        <SelectOption key={subCategory} value={SubCategories[subCategory]}>
+          {subCategory}
+        </SelectOption>
+      );
+    });
+  };
+
+  renderPointsInput = () => {
     const {
       form: { getFieldDecorator },
     } = this.props;
 
+    if (this.state.pointsCompeleted === 'completed') {
+      return (
+        <Col span={8}>
+          <FormItem>
+            {getFieldDecorator('points', {
+              rules: [
+                {
+                  required: true,
+                  message: 'Please enter points value',
+                },
+              ],
+            })(<InputNumber placeholder="points" />)}
+          </FormItem>
+        </Col>
+      );
+    } else {
+      return <Col span={8} />;
+    }
+  };
+
+  renderSecondaryOptions = () => {
+    const { getFieldDecorator } = this.props.form;
+    if (this.state.categoryId === 2) {
```

Comment:
please use the enum to compare rather than the raw value

Reply:
Thanks. Got it